enzyme/Enzyme/MLIR/Passes/EnzymeBatchPass.cpp

```diff
@@ -244,18 +245,22 @@ LogicalResult batchOperation(
   } else {
     // Create new batched function and store in cache
     std::string newFnName = "batched_" + fn.getName().str();
-    newFunc = batchCloneFunction(fn, newFnName, CI.getBatchShape(),
+    newFunc = batchCloneFunction(builder, fn, newFnName, CI.getBatchShape(),
                                  batchedFunctionCache);
     if (!newFunc) {
       return failure();
     }
   }
 
-  OpBuilder builder(CI);
-  auto dCI = builder.create<func::CallOp>(
-      CI.getLoc(), newFunc.getName(), newFunc.getResultTypes(), CI.getInputs());
-  CI.replaceAllUsesWith(dCI);
-  CI->erase();
+  {
+    IRRewriter::InsertionGuard insertGuard(builder);
+    builder.setInsertionPoint(CI);
+    auto dCI =
+        builder.create<func::CallOp>(CI.getLoc(), newFunc.getName(),
```

Comment:
this should likely just become replaceOp [both replacealluses with and op->erase are invalid]